platforms/nuttx/src/bootloader/microchip/mpfs/main.c

```diff
@@ -372,8 +459,20 @@ flash_func_erase_sector(unsigned sector)
 
 #endif
 
-	uint32_t *addr = (uint32_t *)((uint64_t)APP_LOAD_ADDRESS + sector * ss);
+#if defined(CONFIG_MMCSD)
+	/* Flashing into eMMC/SD */
+
+	/* Erase the whole file when erasing the first sector */
+	if (sector == 0) {
+		create_px4_file();
+	}
+
+#endif
+
+	/* Erase the RAM contents */
+
 	memset(addr, 0xFFFFFFFF, ss);
```

Comment:
Is there some not obvious reason from memsetting RAM to all 1's ? Seems like some funny remnant of flash erase.

Reply:
Yes ;) in common code it "verifies" that the erase succeeded by reading back and checking that it is ffff. Doesn't make much sense with RAM but keeps that part happy...